### DISCUSSION TITLE
[Question] Creating FAP from scratch

### DISCUSSION THREAD

**Discussion by unrealsolver** [dcoles/flipperzero-rs#52] [Q&A] [answered]
Hello, I am trying to create a custom FAP using flipperzero-rs crate. I am new to Rust and not sure what I am doing wrong.
I created a new crate with following `Cargo.toml`
```toml
[package]
name = "pms"
version = "0.1.0"
edition = "2021"

[dependencies]
flipperzero = "0.7.2"
flipperzero-rt = "0.7.2"
flipperzero-sys = "0.7.2"
```

And when I run `cargo build --release` I get this:
```
ruslan@nba-153-01a-UZ ~/d/pms (master)> cargo build --release
   Compiling flipperzero-rt v0.7.2
   Compiling flipperzero-sys v0.7.2
   Compiling flipperzero v0.7.2
   Compiling pms v0.1.0 (/home/ruslan/dev/pms)
error: language item required, but not found: `eh_personality`
  |
  = note: this can occur when a binary crate with `#![no_std]` is compiled for a target where `eh_personality` is defined in the standard library
  = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`

error: could not compile `pms` due to previous error
```

And if I use this command `cargo build --target thumbv7em-none-eabihf` I get really long error message which complains for `rust-lld: error: undefined symbol: gpio_ext_pc0`

How do I should modify Cargo.toml / command to build the app?

**Comment by unrealsolver** [comment, marked as answer]
**Answer**: I figured out, that I have to copy `crates/.cargo` to my project root to build the app.
**New questions**: Am I doing it right? Should I add it to your README.md?

**Reply by dcoles** [reply]
Hi @unrealsolver,

That's great to hear! Glad that you got it working. Yes, that's correct. You need `.cargo/config.toml` in your project to define a few key values including `[build] target = target = "thumbv7em-none-eabihf"`.

We had [a bunch of nice example projects](https://github.com/flipperzero-rs/flipperzero/tree/v0.7.2/examples), but they were removed from `main` since we're in the process of moving the examples to their own repository. Though I forgot that this meant that it would no longer be shown on the current `main`. Oops.

Sorry that it made it unclear how to get started.

**Reply by dcoles** [reply]
I believe you also need to remove the [`path = "../../crates/…` lines from `Cargo.toml`](https://github.com/flipperzero-rs/flipperzero/blob/v0.7.2/examples/hello-rust/Cargo.toml#L20-L22).

**Reply by dcoles** [reply]
There is now a [`flipperzero-template`](https://github.com/flipperzero-rs/flipperzero-template) project to help folks get started.
